01_Data/src/layers/sequelize/repository/DeviceModel.ts

```diff
@@ -65,15 +64,28 @@ export class SequelizeDeviceModelRepository extends SequelizeRepository<Variable
     const [component, variable] = await this.findOrCreateEvseAndComponentAndVariable(value.component, value.variable, stationId);
 
     let dataType: DataEnumType | null = null;
+
     if (value.variableCharacteristics) {
-      const [variableCharacteristics, _variableCharacteristicsCreated] = await this.variableCharacteristics.upsert(
-        VariableCharacteristics.build({
-          ...value.variableCharacteristics,
-          variable,
-          variableId: variable.id,
-        }),
-      );
-      dataType = variableCharacteristics.dataType;
+      dataType = value.variableCharacteristics.dataType;
+      const vc = VariableCharacteristics.build({
+        ...value.variableCharacteristics,
+        variableId: variable.id,
+      });
+      console.log(`variableCharacteristics: ${JSON.stringify(vc)}`);
+      await this.s.transaction(async (transaction) => {
+        const savedVariableCharacteristics = await this.s.models[VariableCharacteristics.MODEL_NAME].findOne({
+          where: {
+            variableId: variable.id,
+          },
+          transaction,
+        });
+
+        if (!savedVariableCharacteristics) {
+          await vc.save({ transaction });
```

Comment:
Same emit question here. If we need emit, then we might use CrudRepository here as well as for update.
```suggestion
          await this.variableCharacteristics.create(vc);
```